includes/shortcodes/class-wc-shortcode-checkout.php

```diff
@@ -153,7 +153,7 @@ private static function order_pay( $order_id ) {
 							$held_stock     = ( $hold_stock_minutes > 0 ) ? wc_get_held_stock_quantity( $product, $order->get_id() ) : 0;
 							$required_stock = $quantities[ $product->get_stock_managed_by_id() ];
 
-							if ( $product->get_stock_quantity() < ( $held_stock + $required_stock ) ) {
+							if ( ! apply_filters( 'woocommerce_pay_order_product_has_enough_stock', ! ( $product->get_stock_quantity() < ( $held_stock + $required_stock ) ), $product, $order ) ) {
```

Comment:
this `!` check keeps consistency with a similar one in L142, where, if found false, would trigger an `Exception` and thus a notice for the customer. The logic within the first argument of the filter would remain unchanged (stock quantity is less than held+required), however for naming consistency I reversed the outcome with `!` bang so it's semantically correct.